toolkit/tools/specreader/specreaderutils/specreaderutil.go

```diff
@@ -0,0 +1,706 @@
+		}
+		defer chroot.Close(leaveFilesOnDisk)
+	}
+
+	buildArch, err := rpm.GetRpmArch(runtime.GOARCH)
+	if err != nil {
+		return
+	}
+
+	doParse := func() error {
+		var parseError error
+
+		if targetArch == "" {
+			packageRepo, parseError = parseSPECs(specsDir, rpmsDir, srpmsDir, toolchainDir, distTag, buildArch, specListSet, toolchainRPMs, workers, runCheck)
+			if parseError != nil {
+				err := fmt.Errorf("failed to parse native specs (%w)", parseError)
+				return err
+			}
+		} else {
+			packageRepo, parseError = parseSPECs(specsDir, rpmsDir, srpmsDir, toolchainDir, distTag, targetArch, specListSet, toolchainRPMs, workers, runCheck)
+			if parseError != nil {
+				err := fmt.Errorf("failed to parse cross specs (%w)", parseError)
+				return err
+			}
+		}
+
+		return parseError
+	}
+
+	if chroot != nil {
+		logger.Log.Info("Parsing SPECs inside a chroot environment")
+		err = chroot.Run(doParse)
+	} else {
+		logger.Log.Info("Parsing SPECs in the host environment")
+		err = doParse()
+	}
+
+	if err != nil {
+		return
+	}
+
+	b, err := json.MarshalIndent(packageRepo, "", "  ")
+	if err != nil {
+		logger.Log.Error("Unable to marshal package info JSON")
+		return
+	}
+
+	err = file.Write(string(b), outputFile)
+	if err != nil {
+		logger.Log.Errorf("Failed to write file (%s)", outputFile)
+		return
+	}
+
+	return
+}
+
+// createChroot creates a chroot to parse SPECs inside of.
+func createChroot(workerTar, buildDir, specsDir, srpmsDir string) (chroot *safechroot.Chroot, err error) {
+	const (
+		chrootName       = "specparser_chroot"
+		existingDir      = false
+		leaveFilesOnDisk = false
+	)
+
+	// Mount the specs and srpms directories to an identical path inside the chroot.
+	// Since specreader saves the full paths to specs in its output that grapher will then consume,
+	// the pathing needs to be preserved from the host system.
+	var extraDirectories []string
+
+	extraMountPoints := []*safechroot.MountPoint{
+		safechroot.NewMountPoint(specsDir, specsDir, "", safechroot.BindMountPointFlags, ""),
+		safechroot.NewMountPoint(srpmsDir, srpmsDir, "", safechroot.BindMountPointFlags, ""),
+	}
+
+	chrootDir := filepath.Join(buildDir, chrootName)
+	chroot = safechroot.NewChroot(chrootDir, existingDir)
+
+	err = chroot.Initialize(workerTar, extraDirectories, extraMountPoints, true)
+	if err != nil {
+		return
+	}
+
+	// If this is not a regular build then copy in all of the SPECs since there are no bind mounts.
+	if !buildpipeline.IsRegularBuild() {
+		dirsToCopy := []string{specsDir, srpmsDir}
+		for _, dir := range dirsToCopy {
+			dirInChroot := filepath.Join(chroot.RootDir(), dir)
+			err = directory.CopyContents(dir, dirInChroot)
+			if err != nil {
+				closeErr := chroot.Close(leaveFilesOnDisk)
+				if closeErr != nil {
+					logger.Log.Errorf("Failed to close chroot, err: %s", err)
+				}
+				return
+			}
+		}
+	}
+
+	return
+}
+
+func FindSpecFiles(specsDir string, specListSet map[string]bool) (specFiles []string, err error) {
```

Comment:
Is this function now changed to be public?

Reply:
Yes, made this public so it can be used in other packages